lib/whitehall/publishing_api/html_attachment_pusher.rb

```diff
@@ -0,0 +1,95 @@
+require 'whitehall/publishing_api'
+
+module Whitehall
+  class PublishingApi
+    class HtmlAttachmentPusher
+      attr_reader :edition, :event
+      def initialize(edition:, event:)
+        @edition = edition
+        @event = event
+      end
+
+      def call
+        send(event) if respond_to?(event) && edition.respond_to?(:html_attachments)
+      end
+
+      def publish
+        current_html_attachments.each do |html_attachment|
+          api.publish_async(html_attachment)
+        end
+        content_ids_to_remove.each do |content_id|
+          api.publish_redirect_async(
+            content_id,
+            Whitehall.url_maker.public_document_path(edition)
+          )
+        end
+      end
+      alias :force_publish :publish
+
+      def update_draft
+        current_html_attachments.each do |html_attachment|
+          api.save_draft_async(html_attachment)
+        end
+      end
+      # we don't care whether this is a translation or the main document, we just send the
+      # correct html attachments regardless.
+      alias :update_draft_translation :update_draft
+
+      def unpublish
+        unpublishing = edition.unpublishing
+        return if unpublishing.nil?
+
+        destination = if unpublishing.redirect?
```

Comment:
`redirect?` is equivalent: https://github.com/alphagov/whitehall/blob/master/app/models/unpublishing.rb#L25-L27